.github/workflows/main_vax-availability-api.yml

```diff
@@ -43,6 +43,9 @@ jobs:
     needs: BuildAndTest
 
     steps:
+    - name: Checkout repo
+      uses: actions/checkout@v2
```

Comment:
That's a good point, and I thought about artifacts when first writing this. However, unless I'm reading this wrong, I believe workflows with a `push` trigger using the `checkout` action will **only** use the commit SHA that triggered that workflow, regardless of what is merged in after in the same branch. I believe this means that the docker build should only continue using the relevant commit, and if there is something that'll fail in the next merged PR while the previous is running, it won't affect previous.

https://docs.github.com/en/actions/reference/events-that-trigger-workflows#push

I believe it also says that on the homepage of the `checkout` action on Github. Please let me know if I'm reading that wrong. I also just tested this on my personal fork by running two workflows, one slightly after the other, with the second one with a garbage Dockerfile:
first - https://github.com/rayraykay/VaxFinder-backend/runs/2558971967?check_suite_focus=true
second, no good - https://github.com/rayraykay/VaxFinder-backend/runs/2558956591?check_suite_focus=true

We could still stick to using artifacts, but I thought it was a bit strange since we would essentially be checking in the entire repo. I believe artifacts are more meant to be used for output files and binaries.